File: prototype/python/myprototype.py
from abc import ABC, abstractmethod

class ProtoType(ABC):
    def __init__(self, name):
        self.name = name
    def clone(self):
        pass
        
class FruitProtoType(ProtoType):
    def clone(self):
        return FruitProtoType("Cloned Fruit")
    
class VegetableProtoType(ProtoType):
    def clone(self):
        return VegetableProtoType("Cloned Vegetable")

class ProtoTypeFactory(ABC):
    prototypes = []
    def createProtoType(self):
        self.prototypes.append(FruitProtoType("fruit"))
        self.prototypes.append(VegetableProtoType("vegetable"))
    
    def getProtoTypes(self):
        return self.prototypes
        
def main():
    ptf = ProtoTypeFactory()
    ptf.createProtoType()
    
    prototypes = ptf.getProtoTypes()
    
    for prototype in prototypes:
        print(prototype.name)
        
if __name__ == "__main__":
    main()

File: prototype/swift/myprototype.swift
protocol Prototype {
    func clone() -> Prototype
}

struct FruitPrototype: Prototype {

    func clone() -> Prototype {
        return FruitPrototype()
    }
}

struct VegetablePrototype: Prototype {

    func clone() -> Prototype {
        return VegetablePrototype()
    }
}

struct PrototypeFactory {
    static func getPrototype(type: String) -> Prototype? {
        if type == "Fruit" {
            return FruitPrototype()
        } else if type == "Vegetable" {
            return VegetablePrototype()
        }
        return nil
    }
}   
var fruitPrototype = PrototypeFactory.getPrototype(type: "Fruit")
var clonedFruitPrototype = fruitPrototype?.clone()
print("Cloned Fruit Prototype: \(clonedFruitPrototype!)")

var vegetablePrototype = PrototypeFactory.getPrototype(type: "Vegetable")
var clonedVegetablePrototype = vegetablePrototype?.clone()
print("Cloned Vegetable Prototype: \(clonedVegetablePrototype!)")   


